sqlalchemy-stubs/orm/session.pyi

```diff
@@ -123,15 +381,9 @@ class SessionTransaction:
         self, _capture_exception: bool = ..., _to_root: bool = ...
     ) -> Optional[SessionTransaction]: ...
     def close(self, invalidate: bool = ...) -> None: ...
-    def __enter__(self) -> SessionTransaction: ...
+    def __enter__(self: _TSessionTransaction) -> _TSessionTransaction: ...
     def __exit__(self, type_: Any, value: Any, traceback: Any) -> None: ...
 
-class _IdentityMap:
-    @overload
-    def __get__(self, instance: None, owner: Any) -> None: ...
-    @overload
-    def __get__(self, instance: object, owner: Any) -> WeakInstanceDict: ...
-
 class Session(_SessionClassMethods):
```

Comment:
why this is not a subclass of the session protocol above?

Reply:
Also this is missing come methods, like get. Since `_SessionClassMethods` is also not a subclass of the protocol

Reply:
Technically, `Session` doesn't need to inherit from `_SessionProtocol`. It has all of the methods to meet the contract of `_SessionProtocol` (I just checked) and having it inherit from `_SessionProtocol` in the stubs while not in reality would mean type checkers would have a different MRO.

Reply:
I was mistaken wrt `get` I guess it was hidden by github.

My suggestion was just to avoid repeating the definition of the methods. but since that's already done, that's fine either way